data/fs/encoding.go

```diff
@@ -0,0 +1,132 @@
+package fs
+
+import (
+	"fmt"
+	"skrive/data"
+	"strconv"
+	"strings"
+	"time"
+
+	gonanoid "github.com/matoous/go-nanoid/v2"
+)
+
+func escape(raw string) string {
+	escaped := strings.ReplaceAll(raw, "\\", "\\\\")
+	escaped = strings.ReplaceAll(escaped, ";", "\\;")
+
+	return escaped
+}
+
+func encode(d data.Dose) string {
+	return fmt.Sprintf("%s;%s;%s;%s;%s;", escape(string(d.Id)), escape(strconv.FormatInt(d.Time.Unix(), 10)), escape(d.Quantity), escape(d.Substance), escape(d.Route))
+}
+
+const (
+	ENDS_WITH_ESCAPE = iota
+	BAD_TIME
+	UNKNOWN_HEADER
+)
+
+type DecodeError struct {
+	Kind    int
+	context *string
+}
+
+func (e DecodeError) Error() string {
+	switch e.Kind {
+	case ENDS_WITH_ESCAPE:
+		return "The raw string ends with an escape character."
+	case BAD_TIME:
+		return "Failed to parse a time stamp."
```

Comment:
The more common form is "timestamp".
```suggestion
		return "Failed to parse a timestamp."
```